### PR TITLE
"Show in Tree" only works if sorting is alphabetical

### LINKED ISSUE #8476: "Show in Tree" only works if sorting is alphabetical
Hi guys, 

when I am on an object and I run the "Show in Tree" function, it only works if the objects are sorted alphabetically. Is that the way it is supposed to be?

Best regards

Pimcore-Version: 6.8.10

_Originally posted by @polymed in https://github.com/pimcore/pimcore/discussions/8474_

### PR DESCRIPTION
**Pull request by weisswurstkanone**
resolves #8476

### CODE REVIEW

File: bundles/CoreBundle/Resources/translations/en.json [hunk 1]
```diff
@@ -938,5 +938,6 @@
   "preview_item": "Preview Item",
   "version_currently_saved_in_database": "Currently saved in database (but not published)",
   "preview_item": "Preview Item",
-  "deprecated_feature": "<p style='color: orange;'>Deprecated feature. Use %s<br>instead</p>"
+  "deprecated_feature": "<p style='color: orange;'>Deprecated feature. Use %s<br>instead</p>",
+  "reindex_warning": "Chaning it from alphabatical sort order to index will reindex all item. Do you want to continue?"
```
Comment (blankse) [inline bundles/CoreBundle/Resources/translations/en.json]:
```suggestion
  "reindex_warning": "Changing it from alphabetical sort order to index will reindex all item. Do you want to continue?"
```

Comment (jdreesen) [reply bundles/CoreBundle/Resources/translations/en.json]: Additionally:
```suggestion
  "reindex_warning": "Chaning it from alphabatical sort order to index will reindex all items. Do you want to continue?"
```

File: bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php [hunk 2]
```diff
@@ -952,8 +952,27 @@ public function changeChildrenSortByAction(Request $request)
     {
         $object = DataObject::getById($request->get('id'));
         if ($object) {
-            $object->setChildrenSortBy($request->get('sortBy'));
-            $object->setChildrenSortOrder($request->get('childrenSortOrder'));
+            $sortBy = $request->get('sortBy');
+            $sortOrder = $request->get('childrenSortOrder');
+
+            $currentSortBy = $object->getChildrenSortBy();
+            $currentSortOrder= $object->getChildrenSortOrder();
```
Comment (dvesh3) [inline bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php]: unused var `$currentSortOrder`? should we pass this in `reindexBasedOnSortOrder()`?

File: bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php [hunk 3]
```diff
@@ -1169,6 +1123,126 @@ protected function updateIndexesOfObjectSiblings(DataObject\AbstractObject $upda
         }
     }
 
+
+    /**
+     * @param DataObject\AbstractObject $parentObject
+     * @param string $currentSortOrder
+     */
+    protected function reindexBasedOnSortOrder(DataObject\AbstractObject $parentObject, string $currentSortOrder) {
```
Comment (dvesh3) [inline bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php:1130]: I would move this to model..maybe `AbstractObject` and mark this as internal

File: bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php [hunk 4]
```diff
@@ -1169,6 +1123,126 @@ protected function updateIndexesOfObjectSiblings(DataObject\AbstractObject $upda
...
+            }
+            $latestVersion->save();
+        }
+    }
+
+    /**
+     * @param DataObject\AbstractObject $updatedObject
+     * @param int $newIndex
+     */
+    protected function updateIndexesOfObjectSiblings(DataObject\AbstractObject $updatedObject, $newIndex)
```
Comment (dvesh3) [inline bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php:1189]: I would move this to model

File: bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php [hunk 5]
```diff
@@ -1073,78 +1092,13 @@ public function updateAction(Request $request)
         return $this->adminJson(['success' => $success]);
     }
 
-    /**
-     * @param DataObject\AbstractObject $updatedObject
-     * @param int $newIndex
-     */
-    protected function updateIndexesOfObjectSiblings(DataObject\AbstractObject $updatedObject, $newIndex)
-    {
+    private function executeInsideTransaction(callable $fn) {
```
Comment (dvesh3) [inline bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php:1094]: I would move this to model

File: bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php (hunk 2 above)
Comment (weisswurstkanone) [reply bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php]: 1) thx
2) I wouldn't take the value from the databse but the current value from the admin ui instead

File: bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php (hunk 5 above)
Comment (weisswurstkanone) [reply bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php:1094]: Not directly related, but I would do this in one go. There is even a task for this 
https://github.com/pimcore/pimcore/issues/6906

**Review by dvesh3 — CHANGES_REQUESTED** [review]
exception when changing order to index (manually)
image

**Comment by weisswurstkanone** [comment]
Unable to reproduce.

**Review by dvesh3 — APPROVED** [review]

**Pull request merged**